packages/manager/apps/pci-ai-endpoints/src/components/utils/getUserRole.tsx

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect, useMemo, useContext } from 'react';
+import { ShellContext } from '@ovh-ux/manager-react-shell-client';
+
+interface UseUserInfos {
+  getUserRole(): string;
+}
+
+const useUserInfos = (): UseUserInfos => {
+  const { shell } = useContext(ShellContext);
+  const [role, setRole] = useState<string>('');
+
+  useEffect(() => {
+    shell.environment.getEnvironment().then((env) => {
+      const roles = env.user?.auth?.roles;
+      setRole(roles && roles.length > 0 ? roles[0] : '');
+    });
+  }, [shell]);
+
+  const getUserRole = useMemo(() => () => role, [role]);
+
+  return { getUserRole };
+};
```

Comment:
```suggestion
const useUserInfos = (): UseUserInfos => {
  const { shell } = useContext(ShellContext);
  const [role, setRole] = useState<string>('');

  useEffect(() => {
    shell.environment.getEnvironment().then((env) => {
      const roles = env.user?.auth?.roles;
      setRole(roles && roles.length > 0 ? roles[0] : '');
    });
  }, [shell]);

  return { role };
};
```

`useMemo` is useless here.

You cannot use IAM rule to do known if action is possible by the user ?

Reply:
Nope